src/physics/em/detail/SeltzerBergerInteractor.i.hh

```diff
@@ -0,0 +1,145 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2020 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file SeltzerBergerInteractor.i.hh
+//---------------------------------------------------------------------------//
+
+#include "base/ArrayUtils.hh"
+#include "base/Constants.hh"
+#include "random/distributions/BernoulliDistribution.hh"
+#include "random/distributions/GenerateCanonical.hh"
+#include "random/distributions/UniformRealDistribution.hh"
+
+namespace celeritas
+{
+namespace detail
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Construct with shared and state data.
+ *
+ * The incident gamma energy must be at least twice the electron rest mass.
+ */
+SeltzerBergerInteractor::SeltzerBergerInteractor(
+    const SeltzerBergerDeviceRef& device_pointers,
+    const ParticleTrackView&      particle,
+    const Real3&                  inc_direction,
+    const CutoffView&             cutoffs,
+    StackAllocator<Secondary>&    allocate,
+    const MaterialView&           material)
+    : device_pointers_(device_pointers)
+    , particle_id_(particle.particle_id())
+    , inc_energy_(particle.energy())
+    , inc_momentum_(particle.momentum())
+    , inc_direction_(inc_direction)
+    , cutoffs_(cutoffs)
+    , allocate_(allocate)
+    , material_(material)
+{
+    CELER_EXPECT(particle_id_ == device_pointers_.ids.electron
+                 || particle_id_ == device_pointers_.ids.positron);
+    // Check energy range is applicable for this model
+    energy_val_min_
+        = celeritas::min(cutoffs_.energy(device_pointers_.ids.gamma).value(),
+                         inc_energy_.value());
```

Comment:
I think instead of an assertion, we should store the cutoff energy and take the same approach as in `MollerBhabhaInteractor`, where we essentially bypass the interaction if the incident energy is too low. @stognini

Reply:
Yes. Since we agreed that the energy range verification for the model would happen before invoking the interactor, it makes more sense to simply return `Interaction::from_unchanged()` at the beginning of `operator()` if the condition isn't met.